src-tauri/src/demo/analyser/mod.rs

```diff
@@ -469,9 +541,15 @@ impl GameDetailsAnalyser {
                     if let Some(player) = self.get_player_of_entity_mut(&EntityId::from(entity_id)) {
                         match table_name.as_str() {
                             "m_iTeam" => {
-                                player.team = Team::new(
-                                    i64::try_from(&prop.value).unwrap_or_default()
-                                );
+                                let new_team = Team::new(i64::try_from(&prop.value).unwrap_or_default());
+
+                                // "other" team cannot be joined
+                                // it's the default team when a player joins, but that's already
+                                // the default value when a player state is created so there's no
+                                // point in assigning it
+                                if new_team != Team::Other {
+                                    player.team = new_team;
+                                }
```

Comment:
I think this check is unnecessary, and I'm pretty sure it's slower than just assigning.
See https://stackoverflow.com/q/69450139/13118494
```suggestion
                                player.team = new_team;
```

Reply:
Yeah, I added this check when I wasn't sure if a player's team could be overwritten to `Team::Other` (when I was seeing some players without a team in a killfeed).  Turned out it was caused by m_iTeam updates sometimes not coming in, but PlayerTeamEvents seem to have the data promptly

Reply:
And RE: slowness - generally, micro optimizations like this (checking a simple conditional in a loop) aren't worth spending developer time analyzing _unless_ it's been determined to be a significant problem (or your application is incredibly sensitive to loop timing, and you already know you need to squeeze every CPU cycle you can. This is very rare and isn't typically seen outside low-power embedded or realtime hardware).

See https://softwareengineering.stackexchange.com/a/80094

Reply:
I see. I thought this was put in as a micro-optimization :)